src/test/java/co/cask/cdc/plugins/integration/source/CTSQLServerPluginTest.java

```diff
@@ -0,0 +1,410 @@
+package co.cask.cdc.plugins.integration.source;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.etl.api.streaming.StreamingSource;
+import co.cask.cdap.etl.mock.batch.MockSink;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.SparkManager;
+import co.cask.cdc.plugins.common.OperationType;
+import co.cask.cdc.plugins.common.Schemas;
+import co.cask.cdc.plugins.integration.CDCPluginTestBase;
+import co.cask.cdc.plugins.integration.StructuredRecordRepresentation;
+import co.cask.cdc.plugins.source.sqlserver.CTSQLServerConfig;
+import co.cask.hydrator.common.Constants;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.microsoft.sqlserver.jdbc.SQLServerDriver;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@RunWith(Enclosed.class)
+public class CTSQLServerPluginTest extends CDCPluginTestBase {
+  private static final String PLUGIN_NAME = "CTSQLServer";
+  private static final String HOST = "localhost";
+  private static final String PORT = "1433";
+  private static final String USERNAME = "SA";
+  private static final String PASSWORD = "123Qwe123";
+  private static final String DB_NAMESPACE = "dbo";
+
+  private static void createDatabaseWithTracking(String dbName) throws SQLException {
+    try (Connection connection = getConnectionToRoot();
+         Statement statement = connection.createStatement()) {
+      statement.executeUpdate(String.format("create database %s", dbName));
+      statement.executeUpdate(String.format("alter database %s set change_tracking = ON", dbName));
+    }
+  }
+
+  private static void dropDatabaseIfExists(String dbName) throws SQLException {
+    try (Connection connection = getConnectionToRoot();
+         Statement statement = connection.createStatement()) {
+      statement.executeUpdate(String.format("drop database if exists %s", dbName));
+    }
+  }
+
+  private static Connection getConnectionToRoot() throws SQLException {
+    String connectionString = String.format("jdbc:sqlserver://%s:%s", HOST, PORT);
```

Comment:
is this assuming there is a sqlserver running on local host? It would be better to have the host, port, user, and password all configurable through system properties.